portfolio/tail_perfomance.py

```diff
@@ -0,0 +1,277 @@
+            If DataFrame has more than one column, the first column will be used as reference benchmark index.
+
+        freq: str (default = 'Daily')
+            Frequencies available: daily, weekly, monthly 
+        
+        same_window: bool (default = True)
+            If true, it indicates whether the performance metrics are to be calculated for all columns in the index_ts DataFrame over the same time frame.
+            or, just using as much data as possible for each column. 
+
+        window: int (Optional)
+            Window size for fixed window dd calculation. 
+            Defaults to computing unrestricted drawdowns.
+
+        k: int (Optional)
+            Maximum number of drawdowns to use in calculations. 
+            Defaults to using all available drawdowns.
+
+        summary_table: bool (Optional, default = True)
+            If true, it indicates whether the function should return a summary of performance metrics
+            or the performance for each of the drawdowns.
+
+        annualize_ret: bool (default = True)
+            If true, indicates whether returns should be annualized.
+
+    Returns
+    --------
+        Pandas DataFrame with performance in tail risk events of the reference benchmark index containing:
+
+            - Mean reactivity: average performance in tail risk events
+            - Median reactivity: median performance in tail risk events
+            - Reliability: frequency of positive performance in tail risk events
+            - Convexity: median performance minus first quartile performance in tail risk events
+            - Tail beta: conditional beta to reference benchmark index in tail risk events but subtracting the unconditional means instead of conditional means
+            - Avg carry: unconditional average performance
+            - Carry in recovery: average performance during the period of recovery (only calculated for unrestricted drawdown)
+            - Start date: is the start date of the first drawdown
+            - End date: is the end date of the last drawdown
+    """
+
+    logger = logging.getLogger(_name_)
+
+    # Series/DataFrame check
+    if isinstance(index_ts, pd.Series):
+        index_ts = index_ts.dropna().astype(float).to_frame('index_ts')
+    if isinstance(ref_index, pd.Series):
+        ref_index = ref_index.dropna().astype(float).to_frame('ref_index')
+    elif isinstance(ref_index, pd.DataFrame):
+        ref_index = ref_index.iloc[:, [0]].dropna().astype(float)
+    else:
+        msg = "'ref_index' must be a Pandas Series or DataFrame"
+        logger.error(msg)
+        raise TypeError(msg)
+
+    df = pd.concat([ref_index, index_ts], join='outer', axis=1, sort=True).fillna(method='ffill')
+    df = df.dropna(how='any' if same_window else 'all')
+
+    # Volatility, Mean (vol. adjusted)
+    freq_dict = {'daily': 21, 'weekly': 4, 'monthly': 1}
+    vols = np.log(df).diff(freq_dict[freq]).std() * np.sqrt(12.)
```

Comment:
The ann. factor 12. needs to depend on the frequency chosen. Otherwise, it's wrong.